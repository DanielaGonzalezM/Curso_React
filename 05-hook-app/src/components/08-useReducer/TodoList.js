import React from "react";
import PropTypes from "prop-types";
import TodoListItem from "./TodoListItem";

const TodoList = ({ todos, handleToggle, handleDelete }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map((todo, i) => (
        <TodoListItem
          key={todo.id}
          todo={todo}
          index={i}
          handleDelete={handleDelete}
          handleToggle={handleToggle}
        />
      ))}
    </ul>
  );
};
TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToggle: PropTypes.func.isRequired,
};

TodoList.defaultProps = {
  todos: [],
  handleDelete: () => {},
  handleToggle: () => {},
};
export default TodoList;
