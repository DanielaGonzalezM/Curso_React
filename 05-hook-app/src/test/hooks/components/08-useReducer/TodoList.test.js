import {shallow} from 'enzyme';
import TodoList from '../../../../components/08-useReducer/TodoList';
import { demoTodos } from '../../../fixtures/demoTodos';

describe('Pruebas en TodoList', () => { 
    const handleToggle=jest.fn();
    const handleDelete=jest.fn();

    const wrapper=shallow(<TodoList 
        todos={demoTodos}
        handleToggle={handleToggle}
        handleDelete={handleDelete}
    />)

    test('debe de mostrarse correctamente', () => { 
        
        expect(wrapper).toMatchSnapshot();

     });

     test('debe de tener dos todoListItem', () => { 
        
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));

     });

 })