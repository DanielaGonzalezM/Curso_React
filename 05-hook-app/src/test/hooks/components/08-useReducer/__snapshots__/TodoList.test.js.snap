// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TodoList debe de mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender React",
        "donde": false,
        "id": 1,
      }
    }
    todos={Array []}
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender GA",
        "donde": false,
        "id": 2,
      }
    }
    todos={Array []}
  />
</ul>
`;
